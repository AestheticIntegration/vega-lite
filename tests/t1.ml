
open Vega_lite

let () =
  let data =
    Data.inline ~name:"d1" @@
    Values.(
      col2
        "x" Col.(ints @@ Array.init 20 (fun i -> i))
        "y" Col.(floats @@ Array.init 20 (fun i -> float (i mod 3) +. 0.2))
    )
  in
  let mark = Mark.line in
  let viz = Viz.make ~data ~mark () in
  print_endline @@ Viz.to_json_str viz

