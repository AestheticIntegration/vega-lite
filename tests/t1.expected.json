{
  "$schema": "https://vega.github.io/schema/vega-lite/v5.json",
  "mark": "line",
  "encoding": {
    "x": {
      "field": "x",
      "bin": false,
      "type": "quantitative",
      "title": "the xs are here"
    },
    "y": {
      "field": "y",
      "bin": false,
      "type": "quantitative",
      "title": "y no float"
    }
  },
  "data": {
    "values": [
      { "x": 0, "y": 0.2 },
      { "x": 1, "y": 1.2 },
      { "x": 2, "y": 2.2 },
      { "x": 3, "y": 0.2 },
      { "x": 4, "y": 1.2 },
      { "x": 5, "y": 2.2 },
      { "x": 6, "y": 0.2 },
      { "x": 7, "y": 1.2 },
      { "x": 8, "y": 2.2 },
      { "x": 9, "y": 0.2 },
      { "x": 10, "y": 1.2 },
      { "x": 11, "y": 2.2 },
      { "x": 12, "y": 0.2 },
      { "x": 13, "y": 1.2 },
      { "x": 14, "y": 2.2 },
      { "x": 15, "y": 0.2 },
      { "x": 16, "y": 1.2 },
      { "x": 17, "y": 2.2 },
      { "x": 18, "y": 0.2 },
      { "x": 19, "y": 1.2 }
    ],
    "name": "d1",
    "format": null
  }
}
